lingpy/sequence/ngrams.py

```diff
@@ -0,0 +1,685 @@
+# *-* coding: utf-8 *-*
+"""
+This module provides various methods for generating and
+collecting n-grams from sequences. Standard ngrams,
+skip ngrams, and positional ngrams can be collected.
+"""
+
+from itertools import chain, combinations, product
+from functools import partial
+
+def _seq_as_tuple(sequence):
+    """
+    Internal function for automatically converting a
+    string sequence to a tuple, if needed.
+
+    Parameters
+    ----------
+    sequence: list or str
+        The sequence that shall be converted into an
+        iterable.
+
+    Returns
+    -------
+    out: tuple
+        A tuple of the sequence.
+    """
+
+    # We first check for datatype and then for a space,
+    # as the first test is faster (and evaluation is lazy).
+    if isinstance(sequence, str) and ' ' in sequence:
+        return tuple(sequence.split(' '))
+
+    return tuple(sequence)
+
+# This method with zip, besides returning an iterator as desired,
+# is faster than both the previous lingpy implementation and the
+# one in NLTK; as this is the core of the ngram methods, it is
+# important to have at least this primitive as fast as possible.
+# This is intentionally not defaulting to any value for the
+# order, so that users won't confuse a given order to all
+# orders up to and including the given one.
+def get_n_ngrams(sequence, order, pad_symbol='$'):
+    """
+    Build an iterator for collecting all ngrams of a given
```

Comment:
Same as above: one sentence on one line (use backslash if sentence is longer). Afterwards, the longer explanation.

Reply:
All corrected.